seqr/views/apis/anvil_workspace_api_tests.py

```diff
@@ -0,0 +1,135 @@
+WORKSPACE_NAME = 'anvil project name'
+NEW_WORKSPACE_NAME = 'anvil new project'
+
+LOAD_SAMPLE_DATA = [
+    ["Family ID", "Individual ID", "Previous Individual ID", "Paternal ID", "Maternal ID", "Sex", "Affected Status",
+     "Notes", "familyNotes"],
+    ["1", "NA19675", "NA19675_1", "NA19678", "", "Female", "Affected", "A affected individual, test1-zsf", ""],
+    ["1", "NA19678", "", "", "", "Male", "Unaffected", "a individual note", ""],
+    ["21", "HG00735", "", "", "", "Female", "Unaffected", "", "a new family"]]
+
+BAD_SAMPLE_DATA = [["1", "NA19674", "NA19674_1", "NA19678", "NA19679", "Female", "Affected", "A affected individual, test1-zsf", ""]]
+
+REQUEST_BODY = {
+            'genomeVersion': '38',
+            'uploadedFileId': 'test_temp_file_id',
+            'description': 'A test project',
+            'agreeSeqrAccess': True,
+        }
+
+
+@mock.patch('seqr.views.utils.permissions_utils.user_get_workspace_access_level')
+class AnvilWorkspaceAPITest(AnvilAuthenticationTestCase):
+    fixtures = ['users', 'social_auth', '1kg_project']
+
+    @mock.patch('seqr.views.utils.permissions_utils.logger')
+    def test_anvil_workspace_page(self, mock_logger, mock_get_access_level):
+        # Requesting to load data for a non-existing project
+        url = reverse(anvil_workspace_page, args=[WORKSPACE_NAMESPACE, WORKSPACE_NAME])
+        self.check_require_login(url)
+
+        mock_get_access_level.return_value = {"pending": False, "canShare": True, "accessLevel": "WRITER"}
+        response = self.client.post(url)
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(response.url, '/create_project_from_workspace/my-seqr-billing/anvil%20project%20name')
+
+        # Requesting to load data for an existing project
+        url = reverse(anvil_workspace_page, args=[WORKSPACE_NAMESPACE, EXIST_WORKSPACE_NAME])
+
+        mock_get_access_level.return_value = {"pending": False, "canShare": True, "accessLevel": "WRITER"}
+        response = self.client.post(url)
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(response.url, '/project/R0003_test/project_page')
+
+        # Test lack of permissions
+        mock_get_access_level.return_value = {"pending": False, "canShare": False, "accessLevel": "WRITER"}
+        response = self.client.post(url)
+        self.assertEqual(response.status_code, 403)
+        mock_logger.warning.assert_called_with('test_user_no_access does not have sufficient permissions for workspace my-seqr-billing/anvil-project 1000 Genomes Demo')
+
+    @mock.patch('seqr.views.apis.anvil_workspace_api.logger')
+    @mock.patch('seqr.views.apis.anvil_workspace_api.load_uploaded_file')
+    @mock.patch('seqr.views.apis.anvil_workspace_api.add_service_account')
+    @mock.patch('seqr.utils.communication_utils.EmailMessage')
+    @mock.patch('seqr.utils.middleware.logger')
+    def test_create_project_from_workspace(self, mock_logger, mock_email, mock_add_service_account,
```

Comment:
the names `mock_logger` and `mock_local_logger` are confusin, you should name them something clearer like `mock_middleware_logger` and `mock_api_logger`